### PR TITLE
Event

### PR DESCRIPTION
**Pull request by basnetlaxmi**
In this milestone, I :

added logical files
functionality to the calculator using state, props
Readme File

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,84 +1,89 @@
...
-            <button type="button" className="button-column-1">
+            <button type="button" className="button-column-1" onClick={this.handleClick}>
               0
             </button>
 
-            <button type="button" className="button-column orange">
+            <button type="button" className="button-column orange" onClick={this.handleClick}>
               =
             </button>
           </div>
```
Comment (julie-ify) [inline src/components/Calculator.js]: - [ ] Nice work so far 🔥. However, you're missing the `.` button implementation. Kindly add `.` to your calculation and implement its functionality too.

Your Implementation | Supposed Implementation
:---:|:---:
 |

File: README.md [hunk 2]
```diff
@@ -1,70 +1,63 @@
...
+To get a local copy up and running follow these simple example steps.
 
-The build is minified and the filenames include the hashes.\
-Your app is ready to be deployed!
+1) Enter `git@github.com:basnetlaxmi/math-magician.git` into terminal
 
-See the section about [deployment](https://facebook.github.io/create-react-app/docs/deployment) for more information.
+2) cd math-magician directory
+3) npm start
+4) View in the browser
```
Comment (julie-ify) [inline README.md]: - [ ] Kindly include instructions on how to install your project dependencies. So after number `2` which is `cd math-magician directory,` you can add `npm install` before `npm start`

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @basnetlaxmi,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️  
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @basnetlaxmi 👋, 

 ## Required Changes ♻️
I understand that you might have a tight deadline, but I will like to advise that you implement the changes requested by the previous reviewer. The feedback only makes your project better and complete. It also improves the functionality and makes it easier for a recruiter to be impressed with your project. 🙏 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/Calculate.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+/**
+ * Given a button name and a calculator data object, return an updated
+ * calculator data object.
+ *
+ * Calculator data object contains:
+ *   total:s      the running total
+ *   next:String       the next number to be operated on with the total
+ *   operation:String  +, -, etc.
+ */
```
Comment (Whoistolu) [inline src/logic/Calculate.js:15]: - [ ] [OPTIONAL] Kindly remove all unused lines of code in order to have a clean code-base.

File: src/logic/Calculate.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    return {};
+  }
+
+  // Button must be an operation
+
+  // When the user presses an operation button without having entered
+  // a number first, do nothing.
+  // if (!obj.next && !obj.total) {
+  //   return {};
+  // }
```
Comment (Whoistolu) [inline src/logic/Calculate.js:96]: - [ ] [OPTIONAL] Kindly remove all unused lines of code in order to have a clean code-base.

File: src/logic/Calculate.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+  if (buttonName === '=') {
+    if (obj.next && obj.operation) {
+      return {
+        total: operate(obj.total, obj.next, obj.operation),
+        next: null,
+        operation: null,
+      };
+    }
+    // '=' with no operation, nothing to do
```
Comment (Whoistolu) [inline src/logic/Calculate.js:76]: This single line of comment is okay for a description 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @basnetlaxmi 

Great job implementing the recommended changes by the previous reviewers 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**